src/validationinterface.cpp

```diff
@@ -176,12 +165,14 @@ void ThreadNotifyWallets(CBlockIndex *pindexLastTip)
             CBlockIndex *pindex = chainActive.Tip();
             pindexFork = chainActive.FindFork(pindexLastTip);
 
-            // Fetch recently-conflicted transactions. These will include any
-            // block that has been connected since the last cycle, but we only
-            // notify for the conflicts created by the current active chain.
-            recentlyConflicted = DrainRecentlyConflicted();
+            // Iterate backwards over the connected blocks until we have at
+            // most WALLET_NOTIFY_MAX_BLOCKS to process.
+            while (pindex && pindex->nHeight > pindexFork->nHeight + WALLET_NOTIFY_MAX_BLOCKS) {
```

Comment:
For future improvement: a connected block is necessarily a validated block, so we could have counted the outputs while it was being validated.